### PR TITLE
cli `branch list`: list tracked branches

### LINKED ISSUE #3105: FR: `jj branch list --tracked`
**Is your feature request related to a problem? Please describe.**
It has come up in Discord that we have no way of listing just the tracked branches, which is useful information if you use `jj` in a primarily branching oriented workflow.  

**Describe the solution you'd like**
Implement `jj branch list --tracked` to display all tracked branches.

### PR DESCRIPTION
**Pull request by prscoelho**
# Checklist

If applicable:
- [X] I have updated `CHANGELOG.md`
- [X] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [X] I have added tests to cover my changes

Closes #3105

### CODE REVIEW

File: cli/src/commands/branch.rs
Comment (yuja) [inline cli/src/commands/branch.rs:107]: I don't think `--tracked` and the other name filters should conflict. `--tracked=origin foo` is reasonable use case.

fwiw, that also applies to `--all` technically, but `--all [NAME]..` is disabled just because filtered result doesn't sound like "all". iirc, @ilyagr have some idea to unblock it.

File: cli/src/commands/branch.rs
Comment (prscoelho) [reply cli/src/commands/branch.rs:108]: I'm not sure how much more tedious it is. I wasn't particularly happy with `-t remote1 -t remote2` plus this will simplify the code much more!

That said, what do you think about omitting the git remote from `jj branch list --tracked`? Since it's always tracking.. This was the main motivation behind having it require a remote name since it seemed like the most used case would be for the origin remote.

File: cli/src/commands/branch.rs
Comment (prscoelho) [reply cli/src/commands/branch.rs:692]: That is much better!

File: cli/src/commands/branch.rs
Comment (prscoelho) [reply cli/src/commands/branch.rs:108]: Oh, I misread.. I get it, we may want --remote later or just through name. In that case I'll only do tracked as a boolean and filtering different remotes can be done in another commit or just by the standard filtering.

File: cli/src/commands/branch.rs
Comment (prscoelho) [reply cli/src/commands/branch.rs]: But for this change it's not needed, I guess if we end up going with `--remote`?

File: cli/src/commands/branch.rs
Comment (yuja) [reply cli/src/commands/branch.rs]: > That said, what do you think about omitting the git remote from `jj branch list --tracked`? Since it's always tracking..

Suppose the motivation of using `--tracked` is to list tracked (real) remote branches, it's probably okay to filter out the `@git` remote. We already have a similar special case in `remote_branches()` revset.

If we add some way to filter remotes by name, maybe we can say that `--tracked` implies `--remote=!"git"` by default? (and the user-specified remote filter will override the default.)

> we may want --remote later or just through name.

I don't have any concrete plan, but I just thought it might make sense to add `--remote` or `glob:*@<remote>`.

Comment (prscoelho) [reply cli/src/commands/branch.rs]: Sounds good, I went ahead and added the default filtering for != git and it can be easily extended to adapt for --remote later.

File: cli/src/commands/branch.rs
Comment (yuja) [inline cli/src/commands/branch.rs:106]: (here, and in docs) "the git remote" sounds like a remote repository of git backend (i.e. normal remotes.)

We use "Git-tracking branches" in several places instead. I don't know if it's clear enough, but I think it's less confusing.

**Review by yuja — APPROVED** [review]
Thanks!

File: cli/src/commands/branch.rs
Comment (prscoelho) [reply cli/src/commands/branch.rs:106]: What about "Omits local git-tracking branches by default"?

Comment (yuja) [reply cli/src/commands/branch.rs:106]: > Omits local git-tracking branches by default

Much better. Let's use capitalized "Git-tracking" for consistency.

Comment (prscoelho) [reply cli/src/commands/branch.rs]: Done!

Comment (prscoelho) [reply cli/src/commands/branch.rs]: I resolved the discussions since you approved, hopefully that's fine!

Comment (yuja) [reply cli/src/commands/branch.rs]: Sure, feel free to merge it.

Comment (prscoelho) [reply cli/src/commands/branch.rs]: I don't think I have write access to merge it

Comment (yuja) [reply cli/src/commands/branch.rs]: > I don't think I have write access to merge it

^ @martinvonz ?

Comment (martinvonz) [reply cli/src/commands/branch.rs]: Sent an invite. Thanks for the PR! And thanks to Yuya for reviewing!

**Pull request merged**
